src/test/java/uk/gov/hmcts/reform/civil/handler/tasks/EvidenceUploadCheckHandlerTest.java

```diff
@@ -0,0 +1,66 @@
+package uk.gov.hmcts.reform.civil.handler.tasks;
+
+import org.camunda.bpm.client.task.ExternalTask;
+import org.camunda.bpm.client.task.ExternalTaskService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.civil.service.search.EvidenceUploadNotificationSearchService;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class EvidenceUploadCheckHandlerTest {
+
+    @Mock
+    private ExternalTask mockTask;
+
+    @Mock
+    private ExternalTaskService externalTaskService;
+
+    @Mock
+    private EvidenceUploadNotificationSearchService searchService;
+
+    @Mock
+    private ApplicationEventPublisher applicationEventPublisher;
+
+    @InjectMocks
+    private EvidenceUploadCheckHandler handler;
+
+    @BeforeEach
+    void init() {
+        when(mockTask.getTopicName()).thenReturn("test");
+        when(mockTask.getWorkerId()).thenReturn("worker");
+    }
+
+    @Test
+    void shouldEmitEvidenceUploadCheckEvent_whenCasesFound() {
+        long caseId = 1L;
+        Map<String, Object> data = Map.of("data", "some data");
+        List<CaseDetails> caseDetails = List.of(CaseDetails.builder().id(caseId).data(data).build());
+
+        when(searchService.getCases()).thenReturn(caseDetails);
```

Comment:
Minor:
this should be a "given".

`given(searchService.getCases()).willReturn(caseDetails);`

In general, in each test, put three comments:
```// Given
[code for the "given" section of the test: that is, the assumptions around the test]
// When
[code for the "when" section of the test: that is, the call to the method you want to test]
// Then
[code for the "then" section of the test: that is, the assertions you want to verify]
```
In an ideal world we would do BDD tests but as this is not the chosen approach, we need to try and get the best out of the tools we use. Following this approach helps clarifying the structure of the tests and makes them more readable.